### PR TITLE
Fixes changelings retaining fake brains on de-changelining

### LINKED ISSUE #34846: Changelings don't lose their false brains upon losing changeling
Round 82447

An admin gave Nightmare to a traitor as a ling trade

The Nightmare was killed and I removed its brain

The brain that came out was a normal pink human one, rather than brain/nightmare

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #34846

### CODE REVIEW

File: code/game/gamemodes/changeling/powers/regenerate.dm [hunk 1]
```diff
@@ -28,7 +28,9 @@
 			C.regenerate_limbs(1)
 		C.regenerate_organs()
 		if(!user.getorganslot(ORGAN_SLOT_BRAIN))
-			var/obj/item/organ/brain/changeling_brain/B = new()
+			var/obj/item/organ/brain/B = new()
```
Comment (YPOQ) [inline code/game/gamemodes/changeling/powers/regenerate.dm]: Should this be the species’s brain typepath? In case of changeling nightmare/xeno?

Comment (Cyberboss) [reply code/game/gamemodes/changeling/powers/regenerate.dm]: Yeah it should

**Comment by KorPhaeron** [comment]
Thanks

File: code/modules/antagonists/changeling/powers/regenerate.dm [hunk 2]
```diff
@@ -28,7 +28,9 @@
 			C.regenerate_limbs(1)
 		C.regenerate_organs()
 		if(!user.getorganslot(ORGAN_SLOT_BRAIN))
-			var/obj/item/organ/brain/changeling_brain/B = new()
+			var/obj/item/organ/brain/B = new()
```
Comment (vuonojenmustaturska) [inline code/modules/antagonists/changeling/powers/regenerate.dm]: Can you grab a mutant brain type from the species datum instead? Or does that introduce bugs?

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]

**Pull request merged**